### ISSUE TITLE
Batch normalization incompatible with fp16 in Keras

### ISSUE THREAD

**Issue by lminer** [TianzhongSong/keras-FP16-test#3]
Did you do anything special to get this to work with batch normalization in Keras? Whenever I try it I get an error that float16 is not supported. Seems like there's an [issue](https://github.com/keras-team/keras/issues/9582) related to this in the Keras repo.

**Comment by TianzhongSong** [comment]
@lminer Check Keras and TensorFlow verison?

**Comment by lminer** [comment]
Tensorflow is 1.12 and Keras is 2.2.4. I'm using the dataset API so maybe that's the problem.

**Comment by FB-wh** [comment]
> Tensorflow is 1.12 and Keras is 2.2.4. I'm using the dataset API so maybe that's the problem.

I use the same version as you do. Have you solved this problem?

**Comment by lminer** [comment]
@FB-wh No I haven't. Waiting for the next version of tensorflow.

**Comment by FB-wh** [comment]
> @FB-wh No I haven't. Waiting for the next version of tensorflow.
 Golbstein had solved the problem #11605.   I've tried this method, and it works.
https://github.com/Golbstein/KerasExtras/blob/master/BN16.py
